packages/common/src/items/_internal/slugs.ts

```diff
@@ -40,3 +47,42 @@ export function getSlugMaxLength(orgKey: string) {
   const prefix = `${TYPEKEYWORD_SLUG_PREFIX}|${orgKey}|`;
   return TYPEKEYWORD_MAX_LENGTH - prefix.length;
 }
+
+export interface IParsedIdentifier {
+  /* ArcGIS Online item Id */
+  id?: string;
+  /* slug (without org prefix) */
+  slug?: string;
+  /* org key */
+  orgKey?: string;
+}
+
+/**
+ * parse out the item's id, slug, and org key out of an identifier
+ * @param identifier
+ * @returns
+ */
+export const parseIdentifier = (identifier: string): IParsedIdentifier => {
```

Comment:
This seems like it could also work for an Events, where it would return the CUID vs a GUID... Wondering if we can/should hoist this out of /items

Reply:
tbh - I don't know how fetching events work at all, but the great part of this being internal is that we can refactor as needed later.